File: python/file.py
import re

class Data:

    def __init__(self, filename):
        self._filename = filename

    def count(self):
        with open(self._filename, 'r') as f:
            line = f.readline().rstrip()
            m = re.match(r'NUMBER=(\d+)', line)
            return int(m.group(1))

    def get(self, match):
        for line in open(self._filename, 'r'):
            m = re.match(match+':', line)
            if not m:
                continue
            numbers = line[m.end():]
            for n in Data.lazy_split(numbers):
                yield int(n)

    def output(self):
        return self.get('OUTPUT')

    def collections(self):
        count = self.count()
        return [self.get('COLLECTION'+ str(i)) for i in range(count)]

    @staticmethod
    def lazy_split(s, sep=','):
        exp = re.compile(re.escape(sep))
        pos = 0
        while True:
            m = exp.search(s, pos)
            if not m:
                if pos < len(s):
                    yield s[pos:]
                break
            if pos < m.start():
                yield s[pos:m.start()]
            pos = m.end()

File: typescript/src/file.ts
import fs from 'fs';
import readline from 'readline';

export async function getNumber(file: string): Promise<number> {
  const matchNumber = (line: string) => {
    const regex = /^NUMBER=(\d+)$/;
    const match = regex.exec(line);
    if (match) {
      return Number.parseInt(match[1], 10);
    }

    return undefined;
  };

  return new Promise((resolve, reject) => {
    const error = function (error: any) {
      return reject(error);
    };

    const line = function (line: string) {
      const number = matchNumber(line);
      if (number) {
        return resolve(number);
      }

      return reject(new Error('Number not found'));
    };

    const rl = readline.createInterface({
      input: fs.createReadStream(file)
    });
    rl.on('error', error);
    rl.once('line', line);
  });
}

async function * getData(file: string, key: string) {
  const matchLine = (line: string) => {
    const regex = new RegExp(`^${key}:(.+)$`);
    return regex.exec(line);
  };

  const split = function * (string: string) {
    let pos = 0;
    const number = (end?: number) => Number.parseInt(string.slice(pos, end), 10);
    while (true) {
      const idx = string.indexOf(',', pos);
      if (idx === -1) {
        yield number();
        return;
      }

      yield number(idx);
      pos = idx + 1;
    }
  };

  const rl = readline.createInterface({
    input: fs.createReadStream(file)
  });

  for await (const line of rl) {
    const match = matchLine(line);
    if (!match) {
      continue;
    }

    yield * split(match[1]);
  }
}

export const getOutput = (filename: string) => getData(filename, 'OUTPUT');
export const getCollection = (filename: string, number: number) => getData(filename, `COLLECTION${number}`);

